docs/dev/table/sqlClient.md

```diff
@@ -0,0 +1,538 @@
+                                           properties.
+     -h,--help                             Show the help message with
+                                           descriptions of all options.
+     -j,--jar <JAR file>                   A JAR file to be imported into the
+                                           session. The file might contain
+                                           user-defined classes needed for the
+                                           execution of statements such as
+                                           functions, table sources, or sinks.
+                                           Can be used multiple times.
+     -l,--library <JAR directory>          A JAR file directory with which every
+                                           new session is initialized. The files
+                                           might contain user-defined classes
+                                           needed for the execution of
+                                           statements such as functions, table
+                                           sources, or sinks. Can be used
+                                           multiple times.
+     -s,--session <session identifier>     The identifier for a session.
+                                           'default' is the default identifier.
+{% endhighlight %}
+
+{% top %}
+
+### Environment Files
+
+A SQL query needs a configuration environment in which it is executed. The so-called *environment files* define available table sources and sinks, external catalogs, user-defined functions, and other properties required for execution and deployment.
+
+Every environment file is a regular [YAML file](http://http://yaml.org/) that looks similar to the following example. The file defines an environment with a table source `MyTableName` that reads from CSV file. Queries that are executed in this environment (among others) will have a parallelism of 1, an even-time characteristic, and will run in the `table` result mode.
+
+{% highlight yaml %}
+# Define table sources and sinks here.
+
+tables:
+  - name: MyTableName
+    type: source
+    schema:
+      - name: MyField1
+        type: INT
+      - name: MyField2
+        type: VARCHAR
+    connector:
+      type: filesystem
+      path: "/path/to/something.csv"
+    format:
+      type: csv
+      fields:
+        - name: MyField1
+          type: INT
+        - name: MyField2
+          type: VARCHAR
+      line-delimiter: "\n"
+      comment-prefix: "#"
+
+# Execution properties allow for changing the behavior of a table program.
+
+execution:
+  type: streaming
+  time-characteristic: event-time
+  parallelism: 1
+  max-parallelism: 16
+  min-idle-state-retention: 0
+  max-idle-state-retention: 0
+  result-mode: table
+
+# Deployment properties allow for describing the cluster to which table programs are submitted to.
+
+deployment:
+  response-timeout: 5000
+{% endhighlight %}
+
```

Comment:
What is the "General purpose" vs "per-session"??? It should be explained more clearly.